ros2doctor/ros2doctor/api/network.py

```diff
@@ -54,21 +55,26 @@ def category(self):
 
     def check(self):
         """Check network configuration."""
+        result = Result()
         # check ifcfg import for windows and osx users
         try:
             ifcfg_ifaces = ifcfg.interfaces()
         except NameError:
             doctor_warn('ifcfg is not imported. Unable to run network check.')
-            return False
+            result.increment_error()
+            return result
 
         has_loopback, has_non_loopback, has_multicast = _check_network_config_helper(ifcfg_ifaces)
         if not has_loopback:
             doctor_warn('ERROR: No loopback IP address is found.')
+            result.increment_error()
```

Comment:
It seems like this could be made a little simpler for Check classes. The two methods make sense from the perspective of `ros2doctor` where `doctor_warn()` is telling a human there was a problem and `increment_[error|warning]()` is telling code, but `NetworkCheck` doesn't need to know who is using it. What do you think of making the `Result` class responsible for that?

```python3
result.add_error('No loopback IP address is found')
...
result.add_warning('Only loopback IP address is found')
...
result.add_warning('No multicast IP address is found')
```